### ISSUE TITLE
api/:id/wilduri match also api/:id/anything/wilduri 

### ISSUE THREAD

**Issue by TomasVojacek** [gofiber/fiber#1016]
**Question description**
Is path is defined inside group it match also to /group/:id/_anything_/path

curl 127.0.0.1:3005/api/id/anything/wilduri
return 200 and not return 404 as I expect. 

Is this intended?
**fiber version**
v2 v2.1.4
**Code snippet** _Optional_

```go
package main

import (
	"github.com/gofiber/fiber/v2"
)

func main() {
	app := fiber.New()
	app.Get("/normaluri", func(c *fiber.Ctx) error {
		return c.SendString("uri react just to /normaluri")
	})
	group := app.Group("/api/:id")
	group.Get("/wilduri", func(c *fiber.Ctx) error {
		return c.SendString("uri that react to api/*/wilduri")
	})
	app.Listen(":3005")
}
```

**Comment by yashvardhan-kukreja** [comment]
This is not an issue. This is working perfectly the way you programmed it.
So, gofiber looks at `/api/id/anything/wilduri`, like, this /api/**id/anything**/wildurl

Where, in the above URL, :id = id/anything

So, basically.. gofiber will match `/api/anything1/anything2/anything3/wilduri` with `/api/:id/wilduri` where :id=anything1/anything2/anything3

**Comment by yashvardhan-kukreja** [comment]
If you do not want, such kind of endpoint (like, /id/anything or ), you can configure a regex for :id to make sure that it does not include slashes.

For example, (just an example of regex) -> 
"/api/:id(a-zA-Z)"  or "/api/:id(+.)" , etc.

the regex goes inside the parentheses and the :id has to follow the regex otherwise it will give a 404

**Comment by yashvardhan-kukreja** [comment]
@TomasVojacek , if you understand my point, I guess this issue can be closed then.

cc: @Fenny @codemicro

**Comment by Fenny** [comment]
`app.Group` behaves the same as `app.Use` and match path prefixes. So in your example `:id` will act as an wildcard.

**Comment by TomasVojacek** [comment]
From https://docs.gofiber.io/guide/routing
I got feeling that :id will stop after /and :id* turn id to greedy.
Can we extend this part of doc?

**Comment by TomasVojacek** [comment]
I understand now that :id is greedy so id can contain /
But I did not understand solution with regex.
```
	app.Get("/fixed/:id(a-z)/test", func(c *fiber.Ctx) error {
		return c.SendString("fixed uri and id is:" + c.Params("id"))
	})

```
but curl /fixed/abc/test return 404 as well
> If you do not want, such kind of endpoint (like, /id/anything or anything1/anything2/anything3 ), you can configure a regex for :id to make sure that it does not include slashes.
> 
> For example, (just an example of regex) ->
> `/api/:id(a-zA-Z)` or `/api/:id(+.)` , etc.
> 
> the regex goes inside the parentheses and the :id has to follow the regex otherwise it will give a 404

**Comment by Fenny** [comment]
We don't support regex within routes yet, I think he meant to use regex within the handler.

**Comment by ReneWerner87** [comment]
fixed with PR https://github.com/gofiber/fiber/pull/1017

**Comment by Fenny** [comment]
@TomasVojacek fixed in `v2.2.0` 👍 thanks for the report
